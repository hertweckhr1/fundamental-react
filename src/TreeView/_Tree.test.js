import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Tree from './_Tree';
import TreeView from './TreeView';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<Tree />', () => {
    const oneLevelTree = (
        <Tree>
            <TreeView.Item>
                <TreeView.Row>
                    <TreeView.Col>First Level</TreeView.Col>
                </TreeView.Row>
            </TreeView.Item>
        </Tree>
    );

    test('create tree component', () => {
        // one level tree
        let component = renderer.create(oneLevelTree);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Tree component', () => {
            const element = mount(
                <Tree
                    data-sample='Sample'
                    onExpandClick={() => {}} />);

            expect(
                element.find('.fd-tree').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
