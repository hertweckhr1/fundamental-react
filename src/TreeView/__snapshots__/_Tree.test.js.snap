// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tree /> create tree component 1`] = `
<ul
  aria-hidden={false}
  className="fd-tree"
  role="tree"
>
  <li
    aria-expanded={false}
    className="fd-tree__item"
    id="fd-1"
    role="treeitem"
  >
    <div
      className="fd-tree__row"
    >
      <div
        className="fd-tree__col fd-tree__col--control"
      >
        First Level
      </div>
    </div>
  </li>
</ul>
`;
